In [37]:
pip install pyresparser

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [38]:
pip install python-docx

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [39]:
pip install docx

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [40]:
pip install spacy==2.3.1

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [41]:
pip install https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-2.3.1/en_core_web_sm-2.3.1.tar.gz

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 12.0 MB 1.5 MB/s 


In [42]:
pip install nltk

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [43]:
from docx import Document
import nltk

In [44]:
nltk.download('stopwords')

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


True

In [45]:
from pyresparser import ResumeParser
import os
import pandas as pd

In [78]:
resume = input()            # using a sample resume as input

/content/sample_Resume.pdf


In [79]:
# to check extension of a file
# unpacking the tuple
file_name, file_extension = os.path.splitext("".join(resume))

print(file_name)
print(file_extension)


/content/sample_Resume
.pdf


In [60]:
data = ResumeParser(''.join(resume)).get_extracted_data()  # extracting data using resume parser

In [77]:
print(data['skills'])   # printing skills

['Pandas', 'Cloud', 'Matplotlib', 'Pattern', 'Database', 'Mining', 'Sql', 'Training', 'Word', 'Assembly', 'Testing', 'Segmentation', 'Ms excel', 'Spelling', 'Seaborn', 'Excel', 'Algorithms', 'Powerpoint', 'Programming', 'Python', 'Analysis', 'Design', 'Numpy', 'Technical', 'Engineering', 'Mysql']


In [62]:
try:                                    # reading resume based on extension
    doc = Document()
    with open("".join(resume), 'r', encoding='utf-8') as file:
        doc.add_paragraph(file.read())
    doc.save("text.docx")
    data = ResumeParser('text.docx').get_extracted_data()
    print(data['skills'])
except:
    data = ResumeParser("".join(resume)).get_extracted_data()
    print(data['skills'])

['Pandas', 'Cloud', 'Matplotlib', 'Pattern', 'Database', 'Mining', 'Sql', 'Training', 'Word', 'Assembly', 'Testing', 'Segmentation', 'Ms excel', 'Spelling', 'Seaborn', 'Excel', 'Algorithms', 'Powerpoint', 'Programming', 'Python', 'Analysis', 'Design', 'Numpy', 'Technical', 'Engineering', 'Mysql']


In [80]:
doc = Document()                      # reading job description text
with open('/content/ds_job_description.txt', 'r', encoding='utf-8') as file:
  doc.add_paragraph(file.read())
doc.save("job_desc.docx")
job_desc = ResumeParser('job_desc.docx').get_extracted_data()

In [81]:
print(job_desc['skills'])  # sample job description skills

['Python', 'Statistics', 'Seaborn', 'Matlab', 'Numpy', 'Matplotlib', 'Word', 'Testing', 'Nosql', 'R', 'Algorithms', 'Excel']


In [82]:
content = [job_desc['skills'], data['skills']]

In [66]:
from sklearn.feature_extraction.text import CountVectorizer

In [83]:
vectorizer = CountVectorizer(analyzer=lambda x: x)       # implementing count vectorizer
result = vectorizer.fit_transform(content)

In [68]:
from sklearn.metrics.pairwise import cosine_similarity

#  cosine_similarity used to measure document similarity in text analysis.

In [84]:
print('Skillset of a given candidate and reqquired skillset for a job has a similarity of :',cosine_similarity(result)[1][0])

Skillset of a given candidate and reqquired skillset for a job has a similarity of : 0.45291081365783825


The skills mentioned in the above sample resume and sample job description matches with 45 % similarity.

### Conclusion
Using above method a person can shortlist the cadidates, based on the similarity between the required skillset and the candidates actual skillset.

Lets say if the HR wants the similarity criteria of 60 %, any candidates who has atleast 60% of similarity will be selected and others will be filtered.  